connectors/sources/github.py

```diff
@@ -215,20 +369,297 @@ async def validate_config(self):
+                        github_document=repo_object, schema=FILE_SCHEMA
+                    )
+
+                    document["_id"] = f"{repo_name}/{repo_object['path']}"
+                    yield document, repo_object
+                else:
+                    repo_object["_id"] = f"{repo_name}/{repo_object['path']}"
+                    repo_object["_timestamp"] = last_commit_timestamp
+                    yield repo_object, None
+        except UnauthorizedException:
+            raise
+        except Exception as exception:
+            logger.warning(
+                f"Something went wrong while fetching the files of {repo_name}. Exception: {exception}"
+            )
+
+    async def get_content(self, attachment, timestamp=None, doit=False):
+        """Extracts the content for Apache TIKA supported file types.
+
+        Args:
+            attachment (dictionary): Formatted attachment document.
+            timestamp (timestamp, optional): Timestamp of attachment last modified. Defaults to None.
+            doit (boolean, optional): Boolean value for whether to get content or not. Defaults to False.
+
+        Returns:
+            dictionary: Content document with _id, _timestamp and attachment content
+        """
+        attachment_size = int(attachment["size"])
+        if not (doit and attachment_size > 0):
+            return
+
+        attachment_name = attachment["name"]
+        attachment_extension = attachment["extension"]
+
+        if attachment_extension == "":
+            logger.warning(
+                f"Files without extension are not supported by TIKA, skipping {attachment_name}."
+            )
+            return
+
+        if attachment_extension.lower() not in TIKA_SUPPORTED_FILETYPES:
+            logger.warning(
+                f"Files with the extension {attachment_extension} are not supported by TIKA, skipping {attachment_name}."
+            )
+            return
+
+        if attachment_size > FILE_SIZE_LIMIT:
+            logger.warning(
+                f"File size {attachment_size} of file {attachment_name} is larger than {FILE_SIZE_LIMIT} bytes. Discarding file content"
+            )
+            return
+
+        logger.debug(f"Downloading {attachment_name}")
+
+        document = {
+            "_id": f"{attachment['repo_name']}/{attachment['name']}",
+            "_timestamp": attachment["_timestamp"],
+        }
+        file_data = await self.github_client.get_response(resource=attachment["url"])
+        temp_filename = ""
+        async with NamedTemporaryFile(mode="wb", delete=False) as async_buffer:
+            await async_buffer.write(
+                decode_base64_value(content=file_data["content"])  # pyright: ignore
+            )
+            temp_filename = str(async_buffer.name)
+
+        logger.debug(f"Calling convert_to_b64 for file : {attachment_name}")
+
+        await asyncio.to_thread(convert_to_b64, source=temp_filename)
```

Comment:
Should we use TIKA here or will this be part of later work?

Reply:
Actually, at connector level we need to pass base64 content and ingest processor will take care of TIKA